src/custom/components/Menu/index.tsx

```diff
@@ -22,19 +24,113 @@ const Policy = styled(InternalMenuItem).attrs(attrs => ({
 
 const MenuFlyout = styled(MenuFlyoutUni)`
   min-width: 11rem;
+  box-shadow: 0 0 100vh 100vw rgb(0 0 0 / 25%);
+  top: calc(100% + 16px);
+  order: 1;
+
+  ${({ theme }) => theme.mediaWidth.upToMedium`
+    top: initial;
+    bottom: calc(100% + 32px);
+  `}
+
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    top: 0;
+    left: 0;
+    position: fixed;
+    height: 100vh;
+    width: 100vw;
+    border-radius: 0;
+    box-shadow: none;
+    padding: 0;
+    overflow-y: auto;
+  `};
+
+  > a {
+    transition: background 0.2s ease-in-out;
+    display: flex;
+    align-items: center;
+
+    ${({ theme }) => theme.mediaWidth.upToSmall`
+      flex: 0 1 auto;
+      padding: 16px;
+      font-size: 18px;
+        svg {
+          width: 18px;
+          height: 18px;
+          object-fit: contain;
+          margin: 0 8px 0 0;
+        }
+    `};
+
+    > span {
+      display: flex;
+      align-items: center;
+    }
+
+    > span > svg {
+      margin: 0 8px 0 0;
+    }
+  }
+  > a:hover {
+    background: ${({ theme }) => theme.disabled};
+    border-radius: 6px;
+  }
 `
 
 export const Separator = styled(SeparatorBase)`
-  background-color: #0000002e;
+  background-color: ${({ theme }) => theme.disabled};
   margin: 0.3rem auto;
   width: 90%;
+
+  ${({ theme }) => theme.mediaWidth.upToMedium`
+    width: 100%;
+    margin: 24px auto;
+  `};
+`
+
+export const CloseMenu = styled.button`
+  display: grid;
+  justify-content: space-between;
+  align-items: center;
+  background: ${({ theme }) => theme.disabled};
+  border: 0;
+  color: ${({ theme }) => theme.black};
+  height: 36px;
+  position: sticky;
+  top: 0;
+  cursor: pointer;
+  border-radius: 6px;
+  justify-content: center;
+  padding: 0;
+  margin: 0 0 8px;
+
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    height: 48px;
+    border-radius: 0;
+    justify-content: flex-end;
+  `};
+
+  &::after {
+    content: '✕';
+    display: block;
+    font-size: 20px;
+    margin: 0;
+
+    ${({ theme }) => theme.mediaWidth.upToSmall`
+      margin: 0 14px 0 0;
+      font-size: 24px;
+    `};
+  }
 `
 
 export function Menu() {
+  const toggle = useToggleModal(ApplicationModal.MENU)
+
   return (
     <StyledMenu>
       <MenuFlyout>
-        <InternalMenuItem to="/faq">
+        <CloseMenu onClick={toggle} />
```

Comment:
it works, although there's one `useCloseModals`, not sure if it's more appropriate since all u want to do is close

Reply:
Which you prefer? My thinking is `useToggleModal` should work, given the X close icon is only shown when the modal is visible (thus toggling would toggle it off again..).

Reply:
sure, i think is fine. Not sure if is possible, but the other prevents issues if the event get triggered twice. But I would is fine your approach.

Reply:
Yeah, I think your suggestion is safer to prevent any unexpected future issues. Will update.

Reply:
@anxolin Updated now to use the `useCloseModals` hook. My understanding is, this hook doesn't accept any params. and simply dispatches the closing event for all of these:

```
export enum ApplicationModal {
  WALLET,
  SETTINGS,
  SELF_CLAIM,
  ADDRESS_CLAIM,
  CLAIM_POPUP,
  MENU,
  DELEGATE,
  VOTE
}
```

Is that correct? The difference with the toggle hook, was that there you have to explicitly pass an ApplicationModal. Otherwise, I didn't see any issues with the `useCloseModals` hook from brief testing, but just posting here for clarity.

Reply:
Yes, this is the difference: 
![image](https://user-images.githubusercontent.com/2352112/117195023-12d50b00-ade5-11eb-8e6c-e8b016e224ba.png)

Close, will close any modal. No matter which one is opened